src/js/view/controls/controls.js

```diff
@@ -183,14 +183,19 @@ export default class Controls extends Events {
 
         this.div.appendChild(controlbar.element());
 
-        // Settings Menu
+        const localization = model.get('localization');
+        const settingsMenu = this.settingsMenu = SettingsMenu(api, model, this.controlbar, localization);
         let lastState = null;
-        const visibilityChangeHandler = (visible, evt) => {
+
+        this.controlbar.on('menuVisibility', ({ visible, evt }) => {
```

Comment:
`settingsMenu.on('menuVisibility', ({ visible, evt }) => {`?

Why not just listen on `settingsMenu` here and get rid of the event forwarding through the `controlbar` below?

Reply:
The reason I took this route is to avoid passing api and playerViewModel into the menu, and keeping things unidirectional. The innermost parent class that is responsible for both the settings menu and the menu button is controls, so it made sense to put it there.